compiler/parser/parser.go

```diff
@@ -112,6 +112,13 @@ func New(l *lexer.Lexer) *Parser {
 
 // ParseProgram update program statements and return program
 func (p *Parser) ParseProgram() (*ast.Program, *errors.Error) {
+
+	defer func() {
+		if recover() != nil {
+			fmt.Println("Panic has been recover parser ")
```

Comment:
This can be improved to `Parser recovered from panic`

Reply:
thanks!